src/App.js

```diff
@@ -1,5 +1,146 @@
+const { Console, Random } = require('@woowacourse/mission-utils');
+const { MESSAGES, NUMBERS, UNITS, REGEX, CHOICE } = require('./constants');
+
 class App {
-  play() {}
+	computerNumArr = [];
+
+	play() {
+		Console.print(MESSAGES.START);
+		this.initGame();
+		this.playGame();
+	}
+
+	initGame() {
+		this.computerNumArr = [];
+
+		while (this.computerNumArr.length < 3) {
+			let number = Random.pickNumberInRange(1, 9);
+			if (!this.computerNumArr.includes(number))
+				this.computerNumArr.push(number);
+		}
+
+		return this.computerNumArr;
+	}
+
+	playGame() {
+		Console.readLine(
+			MESSAGES.REQUIRE_USER_INPUT(NUMBERS.GAME_MAX),
+			userAnswerStr => {
+				this.isValidNumber(userAnswerStr);
+
+				const [ball, strike] = this.compareComputerWithUser(
+					this.computerNumArr,
+					userAnswerStr
+				);
+				const resultMessage = this.convertScoreToMessage([
+					ball,
+					strike,
+				]);
+
+				this.showResultMessage(resultMessage);
+			}
+		);
+	}
+
+	isValidNumber(userAnswerStr) {
+		const answer = userAnswerStr.replace(REGEX.SPACE, '');
+		const { length } = answer;
+
+		if (length >= 4) throw new Error(MESSAGES.INVALID_LENGTH);
+
+		if (isNaN(answer)) throw new Error(MESSAGES.NOT_A_NUMBER);
+
+		if (this.removeDuplicated(userAnswerStr) !== userAnswerStr)
+			throw new Error(MESSAGES.DUPLICATED_NUM);
+
+		return true;
+	}
+
+	removeDuplicated(s) {
+		let answer = '';
+
+		for (let i = 0; i < s.length; i++) {
+			if (s.indexOf(s[i]) === i) answer += s[i];
+		}
+
+		return answer;
+	}
+
+	isValidChoice(playerChoice) {
+		if (!REGEX.CHOICE.test(playerChoice)) {
+			Console.print(MESSAGES.FORMAT_ERROR_CHOICE);
+			throw Error(MESSAGES.FORMAT_ERROR_CHOICE);
+		}
+		return true;
+	}
+
+	compareComputerWithUser(computerNumArr, userAnswerStr) {
+		let [ball, strike] = [0, 0];
+
+		userAnswerStr.split('').forEach((userAnswerStr, userAnswerIdx) => {
+			this.isValidNumber(userAnswerStr);
+
+			computerNumArr.map((computerNum, computerNumIdx) => {
+				if (
+					computerNum === Number(userAnswerStr) &&
+					computerNumIdx === userAnswerIdx
+				) {
+					strike++;
+					return;
+				}
+				if (computerNum === Number(userAnswerStr)) ball++;
+			});
+		});
+
+		return [ball, strike];
```

Comment:
```suggestion
		return {ball, strike};
```
값을 여러 개 반환할 때는 object destructuring을 쓰는것이 좋습니다! 함수를 호출한 측에서 변수의 순서를 신경쓰지 않아도 되기 때문입니다. airbnb 에도 명시되어 있습니다.

* [airbnb - Use object destructuring for multiple return values](https://github.com/airbnb/javascript#destructuring--object-over-array)

Reply:
당시에는 경황이 없어 못챙겨보다가 리팩토링하면서 리뷰 읽어보고 있습니다.
꼼꼼하고 유익한 리뷰 정말 감사합니다! 👍